packages/react-flex/Spec.md

```diff
@@ -0,0 +1,528 @@
+| `push`   | boolean                                                                                         | `false`       | Defines an auto margin depending on the flex direction. Kept for retro-compatibility but perhaps can be removed. |
+| `basis`  | string (_see [flex-basis](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-basis#values)_) | "`auto`"      | For clarity, renaming of v0 Flex's `size` prop to `basis`.                                                       |
+| `order`  | number                                                                                          | `0`           | Abstraction of `order`.                                                                                          |
+
+#### _Deprecating_
+
+| Original Component | Name            | Comments                                               |
+| ------------------ | --------------- | ------------------------------------------------------ |
+| StackItem          | `disableShrink` | Extraneous prop given the existance of `shrink`.       |
+| StackItem          | `verticalFill`  | Redundant. Same as a style override of `height: 100%`. |
+| FlexItem           | `size`          | Renamed to `basis`.                                    |
+
+## Sample Code
+
+### Basic
+
+```HTML
+<Flex>
+ <span>1</span>
+ <span>2</span>
+</Flex>
+```
+
+### Direction
+
+```HTML
+<Flex direction="column-reverse">
+  <span>bottom</span>
+  <span>top</span>
+</Flex>
+```
+
+### Alignment
+
+```HTML
+<Flex verticalAlign="center">
+  <span>vertically centered</span>
+  <span>vertically centered</span>
+</Flex>
+```
+
+```HTML
+<Flex horizontalAlign="center">
+  <span>horizontally centered</span>
+  <span>horizontally centered</span>
+</Flex>
+```
+
+### Spacing
+
+```HTML
+<Flex gap="5rem">
+  <span>5rem around me</span>
+  <span>5rem around me</span>
+</Flex>
+```
+
+### Wrapping
+
+```HTML
+<Flex wrap>
+  <span>1</span>
+  <span>2</span>
+</Flex>
+```
+
+### Order
+
+```HTML
+<Flex>
+  <span>2</span>
+  <Flex.Item order={1}>1</Flex.Item>
+</Flex>
+```
+
+### Growing
+
+#### Globally
+
+```HTML
+<Flex grow={1}>
+  <span>wide</span>
+  <span>wide</span>
+</Flex>
+```
+
+#### Individually
+
+```HTML
+<Flex>
+  <span>thin</span>
+  <Flex.Item grow={1}>wide</Flex.Item>
+</Flex>
+```
+
+### Shrinking
+
+#### Globally
+
+```HTML
+<Flex shrink={0}>
+  <span>wide</span>
+  <span>wide</span>
+</Flex>
+```
+
+#### Individually
+
+```HTML
+<Flex>
+  <span>thin</span>
+  <Flex.Item shrink={0}>wide</Flex.Item>
+</Flex>
+```
+
+### Complex example
+
+```HTML
+<Flex
+  as="section"
+  direction="row-reverse"
+  horizontalAlign="space-between"
+  verticalAlign="center"
+  gap="5rem"
+  grow={1}
+  wrap inline>
+  <Flex.Item order={3}>3</Flex.Item>
+  <Flex.Item shrink={0}>1</Flex.Item>
+  <Flex.Item basis="25%">2</Flex.Item>
+</Flex>
+```
+
+## Migration
+
+_Describe what will need to be done to upgrade from the existing implementations:_
```

Comment:
It was part of the spec to describe the migration from both, I assume to show what kind of changes it would mean for our users. For the migration guide, I'll be sure to consolidate this info in a more user friendly format!